### PR TITLE
Changes to Core and manager/heartbeat (Fixes #77)

### LINKED ISSUE #77: [BUG] HeartBeat restarting filter when startup time is too long
**Describe the bug**
For some filters, when their starting time is long (more than a second), the socket will not be created right away and the HeartBeat will try to restart the filter.
The consequence is X concurrent filters with identical parameters running, and only one followed by the manager.

**Plateform (please complete the following information):**
 - OS (version): [e.g. FreeBSD (12.0)]
 - Darwin version: [e.g. 1.0]

**To Reproduce**
1. Set a Darwin configuration with Hostlookup filter activated
2. Assign a large (10+MB) db file to Hostlookup (*database* parameter), to slow down startup
3. start Darwin

**Expected behavior**
the Darwin manager and filters should start, but additional Hostlookup filters should start every few seconds, with exactly the same parameters (use of top/htop can help).
If the manager's log level is below ERROR, WARNING logs should be appended in *darwin_manager.log* stating Heartbeat didn't find socket file associated to filter, and trying to restart it.

**Additional context**
Hostlookup filter's database should have a (very) large number of entries to slow down configuration to a couple of seconds

### PR DESCRIPTION
**Pull request by frikilax**
#Changes to Core
- Start Monitor before reading/generating filter configuration
- Close Monitor gracefully before finishing in case of configuration error
- Monitor returns filter's state (starting, configuring, running)

#Changes to manager/heartbeat
- during startup: check that filter is correctly started
- during stop: check that filter is correctly stopped
- removed heartbeat retries on restarting filters (now filters are either correctly started/restarted or faulty)

### CODE REVIEW

File: samples/base/Core.hpp [hunk 1]
```diff
@@ -76,6 +77,8 @@ namespace darwin {
         std::size_t _nbThread;
         std::size_t _cacheSize;
         std::size_t _threshold;
+        boost::atomic<FilterStatusEnum> _filter_status;
```
Comment (HugoSoszynski) [inline samples/base/Core.hpp]: Why not using std::atomic ?

Creating less dependencies to libs might be a good idea.

File: tests/manager_socket/utils.py [hunk 2]
```diff
@@ -147,8 +147,8 @@ def check_filter_files(filter_name, extension=".1"):
 # Responses
 
 RESP_EMPTY     = '{}'
-RESP_ONE       = '{"logs_1": {}}'
-RESP_THREE     = '{"logs_1": {}, "logs_2": {}, "logs_3": {}}'
+RESP_ONE       = '{"logs_1": {"status": "running"}}'
+RESP_THREE     = '{"logs_1": {"status": "running"}, "logs_2": {"status": "running"}, "logs_3": {"status": "running"}}'
```
Comment (HugoSoszynski) [inline tests/manager_socket/utils.py]: Careful merging these.

See #81

Comment (frikilax) [reply tests/manager_socket/utils.py]: yes, I will handle the merging

File: samples/base/Core.hpp (hunk 1 above)
Comment (frikilax) [reply samples/base/Core.hpp]: done

**Pull request merged**